projects/packages/waf/src/class-waf-rules-manager.php

```diff
@@ -212,25 +236,29 @@ public static function generate_rules() {
 
 		// Update the rules file
 		if ( ! $wp_filesystem->put_contents( $entrypoint_file_path, $rules ) ) {
-			throw new \Exception( 'Failed writing rules file to: ' . $entrypoint_file_path );
+			throw new File_System_Exception( 'Failed writing rules file to: ' . $entrypoint_file_path );
 		}
 	}
 
 	/**
 	 * Generates the automatic-rules.php script
 	 *
-	 * @throws \Exception If rules cannot be generated and saved.
+	 * @global \WP_Filesystem_Base $wp_filesystem WordPress filesystem abstraction.
+	 *
+	 * @throws Waf_Exception         If filesystem is unavailable.
+	 * @throws Waf_Exception         If rules cannot be fetched from the API.
+	 * @throws File_System_Exception If file writing fails.
+	 *
 	 * @return void
 	 */
 	public static function generate_automatic_rules() {
-		/**
-		 * WordPress filesystem abstraction.
-		 *
-		 * @var \WP_Filesystem_Base $wp_filesystem
-		 */
 		global $wp_filesystem;
 
-		Waf_Runner::initialize_filesystem();
+		try {
+			Waf_Runner::initialize_filesystem();
+		} catch ( Waf_Exception $e ) {
+			throw $e;
+		}
```

Comment:
```suggestion
		Waf_Runner::initialize_filesystem();
```

We could also just not have the try-catch block here I think?

Reply:
Resolved via https://github.com/Automattic/jetpack/commit/c1d3effd93c8fd23355781bca6b324383c18d498